Two separate buffer overruns found with ASan, crashing

  cbf2nexus -c zlib --list -o i19-1.h5 "${CBF_DATA_INPUT}/1191_00005.cbf"

built with MSVC.  The omission to assign axes appears asymptomatic,
insofar as the tests are concerned.
--- a/src/cbf_binary.c
+++ b/src/cbf_binary.c
@@ -308,7 +308,7 @@
 
   char digest_text [25];
   
-  char byteorder_text [14];
+  char byteorder_text [15];
 
   const char *text;
 
--- a/src/cbf_hdf5.c
+++ b/src/cbf_hdf5.c
@@ -26834,6 +26834,7 @@
                                             cbf_debug_print2("error: %s\n",cbf_strerror(CBF_ALLOC));
                                             error |= CBF_ALLOC;
                                         } else {
+                                            axes = newAxes;
                                             axes[nAxes-1] = axis;
                                         }
                                     }
@@ -26963,7 +26964,7 @@
                                 /* write the offset */
                                 double offset[3] = {0.,0.,0.};
                                 const hsize_t dim[] = {3};
-                                double buf[] = {0};
+                                double buf[3] = {0.,0.,0.};
                                 if (3==axis_settings.nOffset) {
                                     CBF_CALL(cbf_apply_matrix(key->matrix,axis_settings.offset,offset));
                                 }
